contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,1086 @@
+        assert_eq!(Unauthorized {}, result.unwrap_err());
+
+        let env = mock_env();
+        let info = mock_info(ADMIN, &vec![]);
+
+        let result = execute(deps.as_mut(), env, info, msg);
+        // Admin should not be authorized to update the event info
+        assert_eq!(Unauthorized {}, result.unwrap_err());
+    }
+
+    #[test]
+    fn event_info_update_only_if_event_not_started_or_ended() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+            // Add 300 seconds to prevent end time to be already passed
+            end_time: Timestamp::from_seconds(EVENT_END_SECONDS + 300),
+        };
+
+        // Fake current time to event in progress
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS + 100);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventStarted {
+                start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Edge case current time is event start
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventStarted {
+                start_time: Timestamp::from_seconds(EVENT_START_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Fake current time to event ended
+        env.block.time = Timestamp::from_seconds(EVENT_END_SECONDS + 100);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventTerminated {
+                end_time: Timestamp::from_seconds(EVENT_END_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Edge case current time is event end
+        env.block.time = Timestamp::from_seconds(EVENT_END_SECONDS);
+
+        let result = execute(deps.as_mut(), env.clone(), info.clone(), msg.clone());
+        assert_eq!(
+            ContractError::EventTerminated {
+                end_time: Timestamp::from_seconds(EVENT_END_SECONDS),
+                current_time: env.block.time.clone(),
+            },
+            result.unwrap_err()
+        );
+
+        // Current time is before event started
+        env.block.time = Timestamp::from_seconds(EVENT_START_SECONDS - 100);
+
+        execute(deps.as_mut(), env, info, msg).unwrap();
+    }
+
+    #[test]
+    fn invalid_event_info() {
```

Comment:
This test can be handled in `state` or `msg`.

Reply:
I think that is related to contract not to `msg` or `state` since this ensure that the contract correctly discard invalid `EventInfo`  when calling `UpdateEventInfo`.

Reply:
Okay, I see. Can we split it into different tests?
1. invalid msg
2. start time before current time
3. start time eq current time
4. end time before current time
5. end time eq current time